Quarkus JaCoCo extension fails to start Gradle daemon

### Describe the bug

The `io.quarkus:quarkus-jacoco` extension fails to execute when run in an Azure DevOps pipeline using the Gradle wrapper because it is unable to start the Gradle daemon in this environment:

```
...
Caused by: org.gradle.api.GradleException: Could not start Gradle daemon.
Caused by: java.lang.UnsupportedOperationException: Cannot convert relative path ?/.gradle/daemon/7.5.1 to an absolute file.
```

Running `./gradlew build` locally works without any issues and produces the expected code coverage reports, running the same build command in our CI environment fails.

### Expected behavior

The Quarkus JaCoCo extension is able to create a code coverage report in our CI environment as well.

### Actual behavior

Running `./gradlew build` fails with the following exception:

```
...
java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
  [error]: Build step io.quarkus.jacoco.deployment.JacocoProcessor#transformerBuildItem threw an exception: org.gradle.tooling.GradleConnectionException: Could not fetch model of type 'ApplicationModel' using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-7.5.1-bin.zip'.
Caused by: org.gradle.api.GradleException: Could not start Gradle daemon.
Caused by: java.lang.UnsupportedOperationException: Cannot convert relative path ?/.gradle/daemon/7.5.1 to an absolute file.
```

### How to Reproduce?

Add the Quarkus JaCoCo extension to a project und execute `./gradlew build` in an Azure DevOps pipeline.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17 LTS
OpenJDK Runtime Environment SapMachine (build 17.0.6+0-LTS)
OpenJDK 64-Bit Server VM SapMachine (build 17.0.6+0-LTS, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.5.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.5.1
Build time:   2022-08-05 21:17:56 UTC
Revision:     d1daa0cbf1a0103000b71484e1dbfe096e095918

### Additional information

Content of `gradle.properties` file:

```
quarkusPlatformGroupId=io.quarkus.platform
quarkusPlatformArtifactId=quarkus-bom
quarkusPlatformVersion=2.16.5.Final
quarkusPluginId=io.quarkus
quarkusPluginVersion=2.16.5.Final

version=1.0.0
```

--- a/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
+++ b/test-framework/jacoco/deployment/src/main/java/io/quarkus/jacoco/deployment/JacocoProcessor.java
@@ -16,7 +16,6 @@ import org.jacoco.core.runtime.OfflineInstrumentationAccessGenerator;
 import org.jboss.jandex.ClassInfo;
 
 import io.quarkus.bootstrap.model.ApplicationModel;
-import io.quarkus.bootstrap.utils.BuildToolHelper;
 import io.quarkus.bootstrap.workspace.SourceDir;
 import io.quarkus.deployment.ApplicationArchive;
 import io.quarkus.deployment.IsTest;
@@ -105,17 +104,7 @@ public class JacocoProcessor {
             info.classFiles = classes;
 
             Set<String> sources = new HashSet<>();
-            ApplicationModel model;
-            if (BuildToolHelper.isMavenProject(targetdir.toPath())) {
-                model = curateOutcomeBuildItem.getApplicationModel();
-            } else if (BuildToolHelper.isGradleProject(targetdir.toPath())) {
-                //this seems counter productive, but we want the dev mode model and not the test model
-                //as the test model will include the test classes that we don't want in the report
-                model = BuildToolHelper.enableGradleAppModelForDevMode(targetdir.toPath());
-            } else {
-                throw new RuntimeException("Cannot determine project type generating Jacoco report");
-            }
-
+            final ApplicationModel model = curateOutcomeBuildItem.getApplicationModel();
             if (model.getApplicationModule() != null) {
                 addProjectModule(model.getAppArtifact(), config, info, includes, excludes, classes, sources);
             }

